File: miss-ann/reverse/al/al.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x>=0:
            res = int(str(x)[::-1])
            if -2**31 <= res <=2**31-1 :
                return res
            else:
                return 0
        else:
            res = int(str(abs(x))[::-1])#非纯数字组成的字符串转换问题
            if res<=2**31-1:
                return -res
            else:
                return 0

File: miss-ann/reverse/al/al.c
int reverse(int x) {
    long res=0;
    while(x!=0){
        res=res*10+x%10;
        x/=10;
    }
    int flog1=0x7fffffff;
    int flog2=0x80000000;
    if(res<flog2||res>flog1){
        return 0;
    }
   return (int)res; 
    
}
